expression/util.go

```diff
@@ -273,3 +274,77 @@ func Contains(exprs []Expression, e Expression) bool {
 	}
 	return false
 }
+
+// ExtractFiltersFromDNFs checks whether the cond is DNF and extract the same expression in its leaf items and return them as a slice.
+func ExtractFiltersFromDNFs(ctx context.Context, conditions []Expression) []Expression {
+	var allExtracted []Expression
+	for i := len(conditions) - 1; i >= 0; i-- {
+		if sf, ok := conditions[i].(*ScalarFunction); ok && sf.FuncName.L == ast.LogicOr {
+			extracted, remained := extractFiltersFromDNF(ctx, sf)
+			allExtracted = append(allExtracted, extracted...)
+			if remained == nil {
+				conditions = append(conditions[:i], conditions[i+1:]...)
+			} else {
+				conditions[i] = remained
+			}
+		}
+	}
+	return append(conditions, allExtracted...)
+}
+
+// extractFiltersFromDNF extracts the same condition that occurs in every DNF item and remove them from dnf leaves.
+func extractFiltersFromDNF(ctx context.Context, dnfFunc *ScalarFunction) ([]Expression, Expression) {
+	dnfItems := FlattenDNFConditions(dnfFunc)
```

Comment:
If len(dnfItems) == 1, could  we skip the following logic?

Reply:
Or expression cannot have only one leaf.